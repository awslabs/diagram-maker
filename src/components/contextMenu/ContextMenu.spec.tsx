import * as Preact from 'preact';
import { shallow } from 'preact-render-spy';

import { ContextMenu } from '.';

describe('ContextMenu', () => {

  const destroyCallback = jest.fn();
  const renderCallback = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders at the given position and size', () => {
    const contextMenu = shallow(
      <ContextMenu
        position={{ x: 400, y: 600 }}
        renderCallback={renderCallback}
        destroyCallback={destroyCallback}
      />
    );

    expect(contextMenu).toMatchSnapshot();
  });
});
