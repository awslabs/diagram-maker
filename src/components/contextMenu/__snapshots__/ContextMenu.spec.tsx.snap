// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenu renders at the given position and size 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-context-menu"
  style="transform: translate3d(400px, 600px, 0);"
  data-type="DiagramMaker.ContextMenu"
>
  <ComposeView
    renderCallback={[Function mockConstructor]}
    destroyCallback={[Function mockConstructor]}
  >
  </ComposeView>
</div>

`;
